src/Workspaces/Core/Portable/SymbolKey/SymbolKey.SymbolKeyWriter.cs

```diff
@@ -421,7 +421,7 @@ public override object VisitNamedType(INamedTypeSymbol namedTypeSymbol)
                     WriteType(SymbolKeyType.ErrorType);
                     ErrorTypeSymbolKey.Create(namedTypeSymbol, this);
                 }
-                else if (namedTypeSymbol.IsTupleType)
+                else if (namedTypeSymbol.IsTupleType && namedTypeSymbol.TupleUnderlyingType != namedTypeSymbol)
```

Comment:
this warrants documentation.

Reply:
Should we be using the symbol comparer for comparing symbols?

Reply:
This is intentionally comparing as reference. If a tuple has no names, then `TupleUnderlyingType` is `this`, which we can compare as reference.

---
In reply to: [358966957](https://github.com/dotnet/roslyn/pull/39370#discussion_r358966957) [](ancestors = 358966957)

Reply:
I don't see the reference comparison.  Looks like an invocation of operator!= to me.

Reply:
This is in IDE code, which deals with interfaces. See `INamedTypeSymbol` above